feature_engine/variable_manipulation.py

```diff
@@ -62,7 +62,7 @@ def _find_or_check_numerical_variables(
     if isinstance(variables, (str, int)):
         variables = [variables]
 
-    elif not variables:
+    if variables is None or len(variables) == 0:
```

Comment:
If the user passes an empty list, the transformer should raise an error and not select the numerical variables as we would do with this implementation. This would also break backwards compatibility.

I would keep the elif, so that it is checked only if the first condition is not met.

Could we rework?

Reply:
I'm afraid it is a little more complicated than this, and it took me a few hours of debugging. I'll try to explain.

Before changing the logic, neither strings or int were checked for variable type. Also, _empty lists_ were getting processed exactly the same way as None since __not__ means several things like 0, empty string, empty list, None, etc. 

When I tried to change the logic to fix the string/int issue, one of the check_estimator tests broke (specifically, SelectByTargetMeanPerformance) and after some digging I found out that __empty lists were actually getting passed__ at some point during that check. But since an empty list is not None, the second condition was not met anymore and it would get now processed by the else block in the _find_or_check function and throw an error at some point because no numerical (categorical) variables were found in an empty dataframe.

So I decided to fix said bug by adding the empty list condition (len==0). I reiterate that this was the state of things before changing the code, since "not variables" captures both the None case and the [] case. Again, it went unnoticed because there is no test that directly passes an empty list to the _find_or_check functions.

Finally, about the elif:
keeping the elif means that not only the elif block but also the following __else__ block is only executed if the first condition is not met. But that would get us back to str/int not getting checked, because if a str/int is passed then it matches the first condition and it will be converted into a list, but then it won't enter the else block and it won't be checked for typing which is exactly what was happening before,

Reply:
how about this:

1. if isinstance(variables, (str, int)) ==> if it is of type numeric then [variables] else error
2. elif varables is None (instead of not variables): logic as we have now
4. else: first check that it is not empty, then the code we have now with your len change suggestion.

Would this work?

Reply:
I tried to implement the same logic you proposed in issue #67 for the datetime type checker, it works just fine as far as the desired logic goes, but it still breaks the test I mentioned above due to __empty lists being passed__ during that test. Does this qualify as another new issue?

Reply:
could you commit all code you have so far so I can have a look?

Reply:
@solegalli: changing https://github.com/feature-engine/feature_engine/blob/main/feature_engine/selection/target_mean_selection.py#L204 to include="number" incidentally solves this issue without breaking any other test!